<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>udg::QLocalDatabaseConfigurationScreenBase</class>
 <widget class="QWidget" name="udg::QLocalDatabaseConfigurationScreenBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>573</width>
    <height>424</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Local Database Configuration</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../main/main.qrc">
    <normaloff>:/images/icons/database.svg</normaloff>:/images/icons/database.svg</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Database and Cache Location</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="4">
         <widget class="QLabel" name="m_cachePathWarningLabel">
          <property name="text">
           <string>TextLabel</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="m_databasePathWarningIcon">
          <property name="minimumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">image: url(:/images/icons/emblem-warning.svg);</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="textLabel1_9">
          <property name="text">
           <string>Database Location:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="m_textCacheImagePath">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="textLabel1_8">
          <property name="text">
           <string>DICOM Files Cache Location:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QToolButton" name="m_buttonExaminateCacheImage">
          <property name="toolTip">
           <string>Browse disk to find the directory to save the retrieved DICOM files</string>
          </property>
          <property name="icon">
           <iconset resource="../main/main.qrc">
            <normaloff>:/images/icons/document-open.svg</normaloff>:/images/icons/document-open.svg</iconset>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLabel" name="m_databasePathWarningLabel">
          <property name="text">
           <string>TextLabel</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="m_cachePathWarningIcon">
          <property name="minimumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">image: url(:/images/icons/emblem-warning.svg);</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="m_textDatabaseRoot"/>
        </item>
        <item row="0" column="2">
         <widget class="QToolButton" name="m_buttonExaminateDataBase">
          <property name="toolTip">
           <string>Browse disk to find the database</string>
          </property>
          <property name="icon">
           <iconset resource="../main/main.qrc">
            <normaloff>:/images/icons/document-open.svg</normaloff>:/images/icons/document-open.svg</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="m_mustRestartToApplyWarningIcon">
          <property name="minimumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">image: url(:/images/icons/emblem-warning.svg);</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="m_mustRestartToApplyWarningLabel">
          <property name="text">
           <string>Changes on database location will be applied when the application is restarted</string>
          </property>
          <property name="textFormat">
           <enum>Qt::PlainText</enum>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Automatic Cache Management</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="textLabel3_7">
          <property name="text">
           <string>Minimum space required in hard disk to retrieve DICOM objects:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="m_textMinimumSpaceRequiredToRetrieve">
          <property name="maximumSize">
           <size>
            <width>64</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="maxLength">
           <number>4</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="textLabel4_6">
          <property name="text">
           <string>GB</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="m_minimumSpaceWarningIcon">
          <property name="minimumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">image: url(:/images/icons/emblem-warning.svg);</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="m_minimumSpaceWarningLabel">
          <property name="text">
           <string>TextLabel</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <layout class="QHBoxLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QCheckBox" name="m_checkBoxDeletedOldStudies">
          <property name="text">
           <string>Automatically delete studies not viewed in the last</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="m_textMaximumDaysNotViewed">
          <property name="maximumSize">
           <size>
            <width>64</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="maxLength">
           <number>4</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>days</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="m_deleteNotViewedStudiesWarningIcon">
          <property name="minimumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">image: url(:/images/icons/emblem-warning.svg);</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="m_deleteNotViewedStudiesWarningLabel">
          <property name="text">
           <string>TextLabel</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>121</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0">
         <widget class="QCheckBox" name="m_checkBoxDeleteStudiesIfNotEnoughFreeSpaceAvailable">
          <property name="text">
           <string>Delete studies to keep</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="m_freeSpaceWarningIcon">
          <property name="minimumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">image: url(:/images/icons/emblem-warning.svg);</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="m_textSpaceToFreeIfNotEnoughSpaceAvailable">
          <property name="maximumSize">
           <size>
            <width>64</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="maxLength">
           <number>4</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight</set>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLabel" name="m_freeSpaceWarningLabel">
          <property name="text">
           <string>TextLabel</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>GB of free space.</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="5">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>When there is not enough free space to retrieve or import new studies, older studies will be deleted to mantain the specified free space.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="indent">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="0" column="5" rowspan="2">
         <spacer>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>13</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Local Database Maintenance</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="2">
       <widget class="QPushButton" name="m_buttonCompactDatabase">
        <property name="toolTip">
         <string>Compact Database</string>
        </property>
        <property name="text">
         <string>Compact database</string>
        </property>
        <property name="icon">
         <iconset resource="../main/main.qrc">
          <normaloff>:/images/icons/package.svg</normaloff>:/images/icons/package.svg</iconset>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <spacer>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>23</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="m_buttonDeleteStudies">
        <property name="toolTip">
         <string>Delete all studies from the cache</string>
        </property>
        <property name="text">
         <string>Delete all studies</string>
        </property>
        <property name="icon">
         <iconset resource="../main/main.qrc">
          <normaloff>:/images/icons/edit-delete.svg</normaloff>:/images/icons/edit-delete.svg</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>m_textDatabaseRoot</tabstop>
  <tabstop>m_buttonExaminateDataBase</tabstop>
  <tabstop>m_textCacheImagePath</tabstop>
  <tabstop>m_buttonExaminateCacheImage</tabstop>
  <tabstop>m_checkBoxDeletedOldStudies</tabstop>
  <tabstop>m_textMaximumDaysNotViewed</tabstop>
  <tabstop>m_checkBoxDeleteStudiesIfNotEnoughFreeSpaceAvailable</tabstop>
  <tabstop>m_textSpaceToFreeIfNotEnoughSpaceAvailable</tabstop>
 </tabstops>
 <resources>
  <include location="../main/main.qrc"/>
  <include location="../main/main.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_checkBoxDeleteStudiesIfNotEnoughFreeSpaceAvailable</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_textSpaceToFreeIfNotEnoughSpaceAvailable</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>346</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>564</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_checkBoxDeletedOldStudies</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_textMaximumDaysNotViewed</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>303</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>484</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
